pkg/liqoctl/rest/publickey/create.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2019-2023 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package publickey
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/cli-runtime/pkg/printers"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	networkingv1alpha1 "github.com/liqotech/liqo/apis/networking/v1alpha1"
+	"github.com/liqotech/liqo/pkg/liqoctl/completion"
+	"github.com/liqotech/liqo/pkg/liqoctl/output"
+	"github.com/liqotech/liqo/pkg/liqoctl/rest"
+	"github.com/liqotech/liqo/pkg/utils/args"
+)
+
+const liqoctlCreatePublicKeyLongHelp = `Create a PublicKey.
+
+The PublicKey resource is used to define a PublicKey for the external network.
+
+Examples:
+  $ {{ .Executable }} create publickey my-public-key --cluster-id my-cluster-id --type server --gateway-name my-gateway`
+
+// Create creates a PublicKey.
+func (o *Options) Create(ctx context.Context, options *rest.CreateOptions) *cobra.Command {
+	outputFormat := args.NewEnum([]string{"json", "yaml"}, "")
+
+	o.createOptions = options
+
+	cmd := &cobra.Command{
+		Use:     "publickey",
+		Aliases: []string{"publickeys", "publickeies"},
+		Short:   "Create a Public Key",
+		Long:    liqoctlCreatePublicKeyLongHelp,
+		Args:    cobra.ExactArgs(1),
+
+		PreRun: func(cmd *cobra.Command, args []string) {
+			options.OutputFormat = outputFormat.Value
+			options.Name = args[0]
+			o.createOptions = options
+		},
+
+		Run: func(cmd *cobra.Command, args []string) {
+			output.ExitOnErr(o.handleCreate(ctx))
+		},
+	}
+
+	cmd.Flags().VarP(outputFormat, "output", "o",
+		"Output the resulting PublicKey resource, instead of applying it. Supported formats: json, yaml")
+
+	cmd.Flags().StringVar(&o.ClusterID, "cluster-id", "", "The cluster ID of the remote cluster")
+	cmd.Flags().BytesBase64Var(&o.PublicKey, "public-key", nil, "The public key to be used for the Gateway")
```

Comment:
A void array should be more safe?

Reply:
I would leave nil to differentiate between a nil slice and an empty slice. It should be safe as this flag is forwarded as-is to the PublicKey resource, so whoever uses it should handle it as they want.